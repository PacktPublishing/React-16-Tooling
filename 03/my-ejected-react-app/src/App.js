import React, { Component } from 'react';
import './App.css';
import Heading from './Heading';
import Button from './Button';

class App extends Component {
  render() {
    return (
      <div className="App">
        <Heading>
          My App Heading
        </Heading>
        <Button />
      </div>
    );
  }
}

export default App;
