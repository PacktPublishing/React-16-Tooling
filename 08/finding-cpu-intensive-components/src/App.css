.App {
  margin: 3em;
}

.Field {
  display: flex;
  flex-direction: column;
  width: 200px;
}
