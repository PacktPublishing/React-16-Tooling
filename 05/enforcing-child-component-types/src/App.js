// @flow
import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';

import Parent from './Parent';
import ParentWithOneChild from './ParentWithOneChild';
import ParentWithOptionalChild from './ParentWithOptionalChild';
import ParentWithStringOrNumberChild from './ParentWithStringOrNumberChild';
import Child from './Child';

class App extends Component<{}> {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <Parent>
          <Child />
          <Child />
        </Parent>
        <ParentWithOneChild>
          <Child />
        </ParentWithOneChild>
        <ParentWithOptionalChild />
        <ParentWithStringOrNumberChild>
          Child String
        </ParentWithStringOrNumberChild>
      </div>
    );
  }
}

export default App;
