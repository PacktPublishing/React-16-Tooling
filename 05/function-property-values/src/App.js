// @flow
import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import Articles from './Articles';

class App extends Component<{}> {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <Articles />
      </div>
    );
  }
}

export default App;
