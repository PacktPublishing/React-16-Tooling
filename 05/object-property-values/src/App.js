// @flow
import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import Person from './Person';

class App extends Component<{}> {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <Person person={{ name: 'Roger', age:20 }} />
      </div>
    );
  }
}

export default App;
