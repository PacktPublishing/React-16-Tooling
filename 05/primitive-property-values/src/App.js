// @flow
import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import Intro from './Intro';

type Props = {};

class App extends Component<Props> {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <Intro name="React" version={16} />
      </div>
    );
  }
}

export default App;
