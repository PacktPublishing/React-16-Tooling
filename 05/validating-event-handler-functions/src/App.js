// @flow
import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import EventHandler from './EventHandler';

class App extends Component<{}> {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <EventHandler />
      </div>
    );
  }
}

export default App;
