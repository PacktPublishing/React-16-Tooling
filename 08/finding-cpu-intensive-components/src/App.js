import React, { Component } from 'react';
import './App.css';
import Group from './Group';

class App extends Component {
  state = {
    groupCount: 1,
    memberCount: 200,
    groups: []
  };

  refreshGroups = (groups, members) => {
    this.setState(state => {
      const groupCount =
        groups === undefined ? state.groupCount : groups;
      const memberCount =
        members === undefined ? state.memberCount : members;
      return {
        ...state,
        groupCount,
        memberCount,
        groups: new Array(groupCount).fill(null).map((g, gi) => ({
          name: `Group ${gi + 1}`,
          members: new Array(memberCount)
            .fill(null)
            .map((m, mi) => ({ name: `Member ${mi + 1}` }))
        }))
      };
    });
  };

  onGroupCountChange = ({ target: { value } }) => {
    this.refreshGroups(+value);
  };

  onMemberCountChange = ({ target: { value } }) => {
    this.refreshGroups(undefined, +value);
  };

  onAddMemberClick = i => () => {
    this.setState(state => ({
      ...state,
      groups: state.groups.map(
        (v, gi) =>
          i === gi
            ? {
                ...v,
                members: v.members.concat({
                  name: `Member ${v.members.length + 1}`
                })
              }
            : v
      )
    }));
  };

  componentWillMount() {
    this.refreshGroups();
  }

  shouldComponentUpdate(props, state) {
    const totalMembers = ({ groups }) =>
      groups
        .map(({ members: { length } }) => length)
        .reduce((result, m) => result + m);

    return (
      this.state.groupCount !== state.groupCount ||
      this.state.memberCount !== state.memberCount ||
      totalMembers(this.state) !== totalMembers(state)
    );
  }

  render() {
    return (
      <section className="App">
        <div className="Field">
          <label htmlFor="groups">Groups</label>
          <input
            id="groups"
            type="range"
            value={this.state.groupCount}
            min="1"
            max="20"
            onChange={this.onGroupCountChange}
          />
        </div>
        <div className="Field">
          <label htmlFor="members">Members</label>
          <input
            id="members"
            type="range"
            value={this.state.memberCount}
            min="1"
            max="200"
            onChange={this.onMemberCountChange}
          />
        </div>
        {this.state.groups.map((g, i) => (
          <Group
            key={i}
            name={g.name}
            members={g.members}
            onAddMemberClick={this.onAddMemberClick(i)}
          />
        ))}
      </section>
    );
  }
}

export default App;
