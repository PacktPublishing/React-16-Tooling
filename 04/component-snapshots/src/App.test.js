import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

it('renders without crashing', () => {
  const tree = renderer
    .create(<App />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
