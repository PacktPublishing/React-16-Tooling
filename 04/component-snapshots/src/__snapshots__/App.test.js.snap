// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Welcome to React
    </h1>
  </header>
  <p
    className="App-intro"
  >
    To get 
    <em>
      started
    </em>
    , edit 
    <code>
      src/App.js
    </code>
     and save to reload.
  </p>
</div>
`;
